src/transformers/modeling_tf_bart.py

```diff
@@ -780,6 +793,57 @@ def call(self, input_ids: tf.Tensor, use_cache=False):
         return super().call(positions + self.offset)  # super object is not callable for some reason
 
 
+class TFSinusoidalPositionalEmbedding(TFSharedEmbeddings):
```

Comment:
Why are the positional embeddings `TFSharedEmbeddings`? These weights don't seem to be shared if I read the code correctly, *i.e.* both the encoder and decoder instantiate position ids, but as far as I can tell they don't share their weights. I don't think we need the `TFSharedEmbeddings` abstraction here - or maybe there is  something I'm not seeing?

Reply:
great catch will use `TFEmbeddings`.